### PR TITLE
[SPARK-43294][BUILD] Upgrade zstd-jni to 1.5.5-2

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
This pr aims upgrade `zstd-jni` from 1.5.5-1 to 1.5.5-2.

### Why are the changes needed?
New version includes one new improvement `Added support for initialising a "dict" from a direct ByteBuffer`:
- https://github.com/luben/zstd-jni/pull/255

Other changes as follows:
- https://github.com/luben/zstd-jni/compare/v1.5.5-1...v1.5.5-2

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Pass GitHub Actions

### CODE REVIEW

**Review by srowen — APPROVED** [review]
Looks OK, just add context about what the update does

**Comment by LuciferYang** [comment]
Thanks for your review @srowen. I am on vacation and will continue to complete these tasks after May 8th.

________________________________
发件人: Sean Owen ***@***.***>
发送时间: 2023年5月4日 22:57:27
收件人: apache/spark
抄送: yangjie01; Author
主题: Re: [apache/spark] [SPARK-43294][BUILD] Upgrade zstd-jni to 1.5.5-2 (PR #40962)


@srowen approved this pull request.

Looks OK, just add context about what the update does

―

**Comment by srowen** [comment]
The changes are minor:
https://github.com/luben/zstd-jni/compare/v1.5.5-1...v1.5.5-2
I think it's fine.

**Review by dongjoon-hyun — APPROVED** [review]
+1.

Have a nice vacation, @LuciferYang . :)

**Comment by LuciferYang** [comment]
Thanks @dongjoon-hyun @srowen , I am checking the microbenchmark results and updating this pr later

File: core/benchmarks/ZStandardBenchmark-results.txt [hunk 1]
```diff
@@ -2,26 +2,26 @@
...
-Compression 10000 times at level 3 without buffer pool            640            645           3          0.0       64045.8       0.6X
-Compression 10000 times at level 1 with buffer pool               193            197           3          0.1       19278.3       2.0X
-Compression 10000 times at level 2 with buffer pool               244            251           3          0.0       24416.2       1.6X
-Compression 10000 times at level 3 with buffer pool               420            436          10          0.0       42020.0       0.9X
+Compression 10000 times at level 1 without buffer pool            366            470         137          0.0       36624.7       1.0X
+Compression 10000 times at level 2 without buffer pool            412            413           2          0.0       41169.5       0.9X
+Compression 10000 times at level 3 without buffer pool            538            541           3          0.0       53813.4       0.7X
+Compression 10000 times at level 1 with buffer pool               257            259           1          0.0       25702.7       1.4X
```
Comment (LuciferYang) [inline core/benchmarks/ZStandardBenchmark-results.txt:12]: https://github.com/apache/spark/pull/40721#discussion_r1161989573

It seems that this problem has occurred again. I ran `ZStandardBenchmark` on GA 13 times and did not get the same results as before ... @dongjoon-hyun

Comment (LuciferYang) [reply core/benchmarks/ZStandardBenchmark-results.txt:12]: also cc @pan3793

**Comment by dongjoon-hyun** [comment]
Since the one failure seems to be flaky one, we can ignore it.
```
[info] *** 1 TEST FAILED ***
[error] Failed: Total 3638, Failed 1, Errors 0, Passed 3637, Ignored 10, Canceled 2
[error] Failed tests:
[error] 	org.apache.spark.scheduler.HealthTrackerIntegrationSuite
```

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
Ur, @LuciferYang . I tried to merge but this PR seems to have corrupted commits
Screenshot 2023-05-10 at 9 41 35 PM

**Comment by dongjoon-hyun** [comment]
Please make a clean PR.

**Comment by LuciferYang** [comment]
> Please make a clean PR.

OK, Let me make a new one.

**Comment by LuciferYang** [comment]
@dongjoon-hyun I open a new one : https://github.com/apache/spark/pull/41135

**Pull request closed without merging**